src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -85,6 +89,49 @@ class ResultsNeedFullReloadError(RuntimeError):
     """
 
 
+@attr.s(slots=True, hash=False)
+class UQResultMetadata:
+    global_gsa_meta_data: Dict[str, Any] = attr.ib(
+        validator=attr.validators.instance_of(dict)
+    )
+    result_directory: Path = attr.ib(validator=attr.validators.instance_of(Path))
+
+    @classmethod
+    def empty(cls, result_directory: Path) -> Optional["UQResultMetadata"]:
+        return UQResultMetadata(
+            global_gsa_meta_data={}, result_directory=result_directory
+        )
+
+    @classmethod
+    def get_meta_data_from_file(
+        cls, result_directory: Path
+    ) -> Optional["UQResultMetadata"]:
+        """
+        Return the metadata info from result file.
+        :param result_directory:
+            The directory result.
+        """
+        if not result_directory.is_dir():
+            return cls.empty(result_directory=result_directory)
+
+        if not Path(result_directory / "uq_result").is_file():
+            return None
+        else:
```

Comment:
Agree this this API as is is wonky.

I said previously that we might want to raise an error in case the path does not exist, but I'm not against doing like Martin is suggesting:

If the directory or file does not exist, return an empty metadata, otherwise read and return it.